Client/Frontend/Browser/FaviconManager.swift

```diff
@@ -50,11 +54,13 @@ class FaviconManager : BrowserHelper {
                                 return
                             }
                             self.browser?.favicons.append(fav)
-                            self.profile.favicons.addFavicon(fav, forSite: site)
+                            if !(self.browser?.isPrivate ?? false) {
```

Comment:
This might arrive seconds later. I think you probably want to use the captured `isPrivate` as suggested above.